### PR TITLE
Fix utf8 in json, version and partitions

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: test/cache.py [hunk 1]
```diff
@@ -59,6 +59,16 @@ def test_lmod_conf(self):
             'dir': './test/data',
         })
 
+    def test_versions(self):
+        data = [
+            [
```
Comment (itkovian) [inline test/cache.py:64]: Make this a tuple, should not be a list.

File: test/cache.py [hunk 2]
```diff
@@ -59,6 +59,16 @@ def test_lmod_conf(self):
...
+                ['v2106-foss-2021a-highopt', '8-foss-2020b-20210316', '8-intel-2020b', '8-foss-2020b', 'v2106-foss-2021a', '9-foss-2021a'],
+                ['v2106-foss-2021a-highopt', 'v2106-foss-2021a', '9-foss-2021a', '8-intel-2020b', '8-foss-2020b-20210316', '8-foss-2020b'],
```
Comment (itkovian) [inline test/cache.py:66]: I do not grok the sorting here, especially not the sorting of the two last items.

File: lib/vsc/modules/cache.py [hunk 3]
```diff
@@ -48,6 +49,30 @@
...
+# very dumb way to deal with difficulty to get utf8 safe data out of
+# lua using print. issue might also be with run.async
+SIMPLE_UTF_FIX_REGEX = re.compile(r"\\x")
+
+
+class SoftwareVersion(LooseVersion):
+    """Support even weirder non-sensical version schemes"""
+    component_re = re.compile(r'(v?\d+ | [a-z]+ | \.)', re.VERBOSE)
+
+    def parse (self, vstring):
```
Comment (wdpypere) [inline lib/vsc/modules/cache.py:61]: no space after parse, that is now an actual pylint error.

File: test/cache.py (hunk 2 above)
Comment (stdweird) [reply test/cache.py:66]: pick any order, it's in any case much better than the current raised exception ;)

**Pull request merged**